File: app/src/main/java/com/comptechnoid/gads2020leaderboard/adapters/LearningLeadersAdapter.java
package com.comptechnoid.gads2020leaderboard.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.comptechnoid.gads2020leaderboard.R;
import com.comptechnoid.gads2020leaderboard.service.LearningLeader;

import java.util.List;

public class LearningLeadersAdapter extends RecyclerView.Adapter<LearningLeadersAdapter.ViewHolder> {

    private Context ctx;
    private List<LearningLeader> data;

    public LearningLeadersAdapter(Context ctx, List<LearningLeader> data) {
        this.ctx = ctx;
        this.data = data;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.learnin_leaders_vh, parent, false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.name.setText(data.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        TextView name;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
        }
    }
}

File: app/src/main/java/com/comptechnoid/gads2020leaderboard/adapters/LearningLeadersAdapter.kt
package com.comptechnoid.gads2020leaderboard.adapters

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.comptechnoid.gads2020leaderboard.R
import com.comptechnoid.gads2020leaderboard.service.LearningLeader
import com.squareup.picasso.Picasso

class LearningLeadersAdapter(private val data: List<LearningLeader>) :
    RecyclerView.Adapter<LearningLeadersAdapter.ViewHolder>() {
    private var ctx: Context? = null

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        ctx = parent.context
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.learning_leaders_vh, parent, false)
        return ViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        holder.name.text = data[position].name
        holder.hours.text = data[position].hours + " Learning Hours "
        holder.country.text = data[position].country
        Picasso.with(ctx).load(data[position].badgeUrl).placeholder(R.mipmap.ic_launcher)
            .into(holder.badge)
    }

    override fun getItemCount(): Int {
        return data.size
    }

    inner class ViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var name: TextView
        var country: TextView
        var hours: TextView
        var badge: ImageView

        init {
            name = itemView.findViewById(R.id.name)
            country = itemView.findViewById(R.id.country)
            hours = itemView.findViewById(R.id.hours)
            badge = itemView.findViewById(R.id.badge)
        }
    }

}